### PR TITLE
Fixed issue #19049: Cannot export LSS from survey where theme is set to one extended from 'bootswatch'

### PR DESCRIPTION
**Pull request by gabrieljenik**

### CODE REVIEW

**Review by Shnoulle — APPROVED** [review]
I don't see error code, but unsure on the fix …

**Comment by gabrieljenik** [comment]
@olleharstedt Is there anyone else who can review this? Seems a sensistive matter

File: application/models/TemplateConfig.php [hunk 1]
```diff
@@ -1054,6 +1054,24 @@ public static function formatToJsonArray($oFiled, $bConvertEmptyToString = false
...
+    /**
+     * Convert the theme options to json.
+     * @param array|object $options the filed to convert
+     * @return string  json
+     */
+    public static function convertOptionsToJson($options)
+    {
+        $optionsArray = [];
+        foreach ($options as $option => $optionValue) {
+            // Clean every value from newlines, tabs and blank spaces for options
```
Comment (Shnoulle) [inline application/models/TemplateConfig.php:1066]: It potentially broken some theme…

https://gitlab.com/SondagesPro/SurveyThemes/skelvanilla/-/blob/master/views/options/options_logocolors.twig?ref_type=heads#L301

Can be reviewed on another solution ?

**Comment by Shnoulle** [comment]
This broke feature on skelvanilla if i don't make error.

Disabling new lines on theme options value must  be avoided in my opinion.

File: application/models/TemplateConfig.php (hunk 1 above)
Comment (gabrieljenik) [reply application/models/TemplateConfig.php:1066]: This was already happening on formatToJsonArray.

Can I please ask you to expand on what is being broken?

Comment (Shnoulle) [reply application/models/TemplateConfig.php:1066]: i write _potentially_ the textearea result are used for css style here. Need line feed.

Here : it's when export./import lss ? Must try with current version and see what happen, report the issue if issue exist.

Comment (gabrieljenik) [reply application/models/TemplateConfig.php:1066]: I am not sure why lin feeds were removed. We are just porting the code.

Still, why do you need line feed for CSS?

Comment (Shnoulle) [reply application/models/TemplateConfig.php:1066]: > Still, why do you need line feed for CSS?

It's for simple user who, just to add some css to specific survey. You realy want to remove line feed for simple user after copy survey ?

And here : it's a sample, i really think we must not update data … if data have line feed : data export as lss and imported aftsr must have line feed.

Comment (Shnoulle) [reply application/models/TemplateConfig.php:1066]: > This was already happening on formatToJsonArray.
> Can I please ask you to expand on what is being broken?

Can not test currently with skelvanila on 6
But on 5.X : linefeed respected when export/import lss

I try to create a quick compatible version  …

**Comment by Shnoulle** [comment]
With master with [hacked version of skelvanilla ](https://gitlab.com/SondagesPro/SurveyThemes/skelvanilla/-/commits/dev_6_dontuse)(need an options with linefeed)

Current master : export/import : OK : line feed options still here (with the line feeds)
This commit :  export/import : loose line feed options

[limesurvey_survey_417778.lss.txt](https://github.com/LimeSurvey/LimeSurvey/files/13217847/limesurvey_survey_417778.lss.txt)

Capture d’écran du 2023-10-31 15-49-43

**Review by Shnoulle — CHANGES_REQUESTED** [review]
Need line feed compatibility

**Comment by gabrieljenik** [comment]
> Need line feed compatibility

Do you have a clue why that line feed was there originally?
https://github.com/LimeSurvey/LimeSurvey/pull/3521/files#diff-508c0106b60d82f7d3a33272d04c1272ade4e7cc8eeb81d9f3a5e6ecc0ac1570R1033

File: application/models/TemplateConfig.php (hunk 1 above)
Comment (gabrieljenik) [reply application/models/TemplateConfig.php:1066]: > You realy want to remove line feed for simple user after copy survey ?

I don't want to remove anything.
I think this is a side effect. The code which was there maybe was not working and now it is.

**Comment by Shnoulle** [comment]
> > Need line feed compatibility
> 
> Do you have a clue why that line feed was there originally? https://github.com/LimeSurvey/LimeSurvey/pull/3521/files#diff-508c0106b60d82f7d3a33272d04c1272ade4e7cc8eeb81d9f3a5e6ecc0ac1570R1033

I test : no issue with current master …

No clue…

**Comment by Shnoulle** [comment]
OK, i can test again ? Right ?

(Thank you)

**Comment by Shnoulle** [comment]
> > Need line feed compatibility
> 
> Do you have a clue why that line feed was there originally? https://github.com/LimeSurvey/LimeSurvey/pull/3521/files#diff-508c0106b60d82f7d3a33272d04c1272ade4e7cc8eeb81d9f3a5e6ecc0ac1570R1033

I try to add line feed in config.xml to give example … no way , maybe function used only when read config.xml ?

**Comment by gabrieljenik** [comment]
> > > Need line feed compatibility
> > 
> > 
> > Do you have a clue why that line feed was there originally? https://github.com/LimeSurvey/LimeSurvey/pull/3521/files#diff-508c0106b60d82f7d3a33272d04c1272ade4e7cc8eeb81d9f3a5e6ecc0ac1570R1033
> 
> I try to add line feed in config.xml to give example … no way , maybe function used only when read config.xml ?

If you remove the line feed, things work?

**Comment by gabrieljenik** [comment]
> OK, i can test again ? Right ?
> 
> (Thank you)

Please. Do so. Thanks!

File: application/models/TemplateConfig.php [hunk 2]
```diff
@@ -1054,6 +1054,31 @@ public static function formatToJsonArray($oFiled, $bConvertEmptyToString = false
...
+            // Trim values, as they may be in a new line in the XML. For example:
+            // <sample_option>
+            //      default value
+            // </sample_option>
+            // Also, by casting, inner nodes are eliminated
+            // and only the text value inside the node is obtained
+            $optionsArray[$option] = trim((string) $optionValue);
+        }
+        if (empty($optionsArray)) {
+            return '""';
```
Comment (Shnoulle) [inline application/models/TemplateConfig.php:1077]: Why not '' ?

Comment (Shnoulle) [reply application/models/TemplateConfig.php:1077]: OK json_encoded empty string

**Review by Shnoulle — APPROVED** [review]

**Pull request merged**